rest/config.go

```diff
@@ -814,21 +817,29 @@ func (config *ServerConfig) setupAndValidateDatabases() (errs error) {
 
 // expandEnv replaces $var or ${var} in config according to the values of the
 // current environment variables. The replacement is case-sensitive. References
-// to undefined variables are replaced by an empty string. A default value can
+// to undefined variables will result in an error. A default value can
 // be given by using the form ${var:-default value}.
-func expandEnv(config []byte) []byte {
+func expandEnv(config []byte) (value []byte, errs error) {
 	return []byte(os.Expand(string(config), func(key string) string {
 		if key == "$" {
 			base.Debugf(base.KeyAll, "Skipping environment variable expansion: %s", key)
 			return key
 		}
-		return envDefaultExpansion(key, os.Getenv)
-	}))
+		val, err := envDefaultExpansion(key, os.Getenv)
+		if err != nil {
+			errs = multierror.Append(errs, err)
+		}
+		return val
+	})), errs
 }
 
+// ErrEnvVarUndefined is returned when a specified variable can’t be resolved from the system environment or no
+// default value is supplied in the configuration — startup will be aborted.
+const ErrEnvVarUndefined = "undefined environment variable '${%s}' is specified in the config without default value"
```

Comment:
Seems like somewhere we'd want a typed error to allow for use of `errors.Is`/`errors.As`

```suggestion
type ErrEnvVarUndefined struct {
	key string
}

func (e FleeceDeltaError) Error() string {
	return fmt.Sprintf("undefined environment variable '${%s}' is specified in the config without default value", e.key)
}
```

Reply:
I've considered including this as a typed error initially, but later decided to not go with it since there was special handling by using the errors.Is/errors.As. When revisiting this now, it seems reasonable to include typed error because it can be used in the unit tests as well.

Reply:
What do you mean by special handling? I think generally speaking, all new errors we're creating in SG should be compatible with Is/As for consistency reasons and ease of use more than anything else.